dask/array/overlap.py

```diff
@@ -750,7 +739,7 @@ def coerce_depth_type(ndim, depth):
 
 
 def coerce_boundary(ndim, boundary):
-    default = "reflect"
+    default = "none"
     if boundary is None:
         boundary = default
```

Comment:
It's this if condition that is leading to assertion errors in the trim tests. 

If we do a presumably nonsensical thing, and set `boucndary` here equal to any of `"reflect"`, `"constant"`, `"nearest"`, or `None` then all the tests pass. But when we set it equal to the string value `"none"` (instead of the python `None`) then the assertions below fail. I'm not exactly sure what is going on here. 

```
=========================== short test summary info ============================
FAILED dask/array/tests/test_overlap.py::test_trim_internal - assert ((9, 8, ...
FAILED dask/array/tests/test_overlap.py::test_nearest_overlap - AssertionError:
FAILED dask/array/tests/test_overlap.py::test_some_0_depth - AssertionError:
FAILED dask/array/tests/test_overlap.py::test_depth_equals_boundary_length - ...
FAILED dask/array/tests/test_overlap.py::test_depth_greater_than_boundary_length
======
```

Reply:
Yeah I think we just need to add the boundary kwarg to all the `trim_internal` calls in the tests. I just pushed a commit that does that